### ISSUE TITLE
Materials of archived courses should be freezed/should not change

### ISSUE THREAD

**Issue by messa** [pyvec/naucse.python.cz#214]
Right now courses, for example [runs/2017/pyladies-praha-jaro/info.yml](https://github.com/pyvec/naucse.python.cz/blob/master/runs/2017/pyladies-praha-jaro/info.yml), just reference lessons by their path, for example `beginners/testing`.

What happens when the lesson moves to other directory or even disappears? It would break content of some previous course if someone doesn't dedicate their time to update _all_ such references.

What happens when the lesson changes? I think the content of the old course should be preserved as is, as it was presented and teached.

Since all materials are currently in git on Github, accessible through git and Github APIs, this could be doable for example by creating a new branch after each course ends and material for the archived course could be rendered from the given branch instead of master.

**Comment by messa** [comment]
Looking from very high level, this is the same thing as #175 - the archived courses just would be rendered from branch in the upstream repository in same/similar way as contributed courses from PR branches.

**Comment by encukou** [comment]
Right, the plan is to do this after #175 is implemented. Thanks for saying it explicitly :)

**Comment by mikicz** [comment]
I removed the blocked label from this since #175 is practically finished.

**Comment by hroncok** [comment]
Note that MI-PYT runs are frozen and may serve as a reference.
